client/__tests__/e2e/e2e.test.js

```diff
@@ -31,161 +39,244 @@ afterAll(() => {
+  });
+});
+
+describe("metadata loads", async () => {
+  test("categories and values from dataset appear", async () => {
+    for (const label in data.categorical) {
+      await utils.waitByID(`category-${label}`);
+      const category_name = await utils.getOneElementInnerText(
+        `[data-testid="category-${label}"]`
+      );
+      expect(category_name).toMatch(label);
+      await utils.clickOn(`category-expand-${label}`);
+      const categories = await cxgActions.getAllCategoriesAndCounts(label);
+      expect(Object.keys(categories)).toMatchObject(
+        Object.keys(data.categorical[label])
+      );
+      expect(Object.values(categories)).toMatchObject(
+        Object.values(data.categorical[label])
+      );
+    }
+  });
+
+  test("continuous data appears", async () => {
+    for (const label in data.continuous) {
+      await utils.waitByID(`histogram-${label}`);
+    }
   });
 });
 
-describe("search for genes", () => {
-  test("search for known gene and add to metadata", async () => {
-    await page.goto(appUrlBase);
-    await page.waitForSelector("[ data-testid='gene-search']");
+describe("cell selection", async () => {
+  test("selects all cells cellset 1", async () => {
+    const cell_count = await cxgActions.cellSet(1);
+    expect(cell_count).toBe(data.dataframe.nObs);
+  });
+
+  test("selects all cells cellset 2", async () => {
+    const cell_count = await cxgActions.cellSet(2);
+    expect(cell_count).toBe(data.dataframe.nObs);
+  });
+
+  test("selects cells via lasso", async () => {
+    for (const cellset of data.cellsets.lasso) {
+      const cellset1 = await cxgActions.calcDragCoordinates(
+        "layout-graph",
+        cellset["coordinates-as-percent"]
+      );
+      await cxgActions.drag("layout-graph", cellset1.start, cellset1.end, true);
+      const cell_count = await cxgActions.cellSet(1);
+      expect(cell_count).toBe(cellset.count);
+    }
+  });
+
+  test("selects cells via categorical", async () => {
+    for (const cellset of data.cellsets.categorical) {
+      await utils.clickOn(`category-expand-${cellset.metadata}`);
+      await utils.clickOn(`category-select-${cellset.metadata}`);
+      for (const val of cellset.values) {
+        await utils.clickOn(
+          `categorical-value-select-${cellset.metadata}-${val}`
+        );
+      }
+      const cell_count = await cxgActions.cellSet(1);
+      expect(cell_count).toBe(cellset.count);
+    }
+  });
+
+  test("selects cells via continuous", async () => {
+    for (const cellset of data.cellsets.continuous) {
+      const hist_id = `histogram-${cellset.metadata}-plot-brush`;
+      const coords = await cxgActions.calcDragCoordinates(
+        hist_id,
+        cellset["coordinates-as-percent"]
+      );
+      await cxgActions.drag(hist_id, coords.start, coords.end);
+      const cell_count = await cxgActions.cellSet(1);
+      expect(cell_count).toBe(cellset.count);
+    }
+  });
+});
+
+describe("gene entry", async () => {
```

Comment:
how much do you want to test error handling (at least in the common cases)?   Seems like it might be worthwhile adding a test for a) a known bad gene name, b) a known badly formatted bulk gene list (eg,   "foo;bar;baz") and ensure a toast/error appears.

Reply:
@colinmegill didn't like checking on the toasts themselves. I also thought we are likely moving to error collapsing soon so I wasn't sure that this behavior was stable enough to test. What do you think?

Reply:
fair point about the toasts.

But knowing that bad data at least doesn't generate an error on the console is worth something....    There are not many places we accept user data entry - might as well have some "bad data" tests for them - ie, just confirm that we seem to survive the bad data.

Short term, this may require dismissing toasts, so I'm OK if you merge this without adding these tests.  We can augment the tests as the toasts stabilize.